0014-fungible-token-standard.md

```diff
@@ -2,85 +2,178 @@
 
+Emits a `BurnEvent` (see [Events](#events)).
+
+#### Query Balances
 ```TypeScript
-interface Mintable {
-  totalSupply: State<UInt64>;
-  circulatingSupply: State<UInt64>;
-  mint: (to: PublicKey, amount: UInt64) => AccountUpdate;
-  setTotalSupply: (amount: UInt64) => void;
-}
+@method.returns(UInt64) async getBalanceOf(address: PublicKey)
+```
+
+Returns the balance of the token account for the public key `address`.
+
+### Privileged Administrative functions
+
+In this section, we list methods that require some sort of special privileges, like minting new tokens. The rules for when it is permissible to mint new tokens will be different for different tokens. A simple rule could be to require a signature from one or more special keys. There could also be a total limit on the number of tokens in existence, or it could be forbidden to mint new tokens at all.
 
-interface Burnable {
-  burn: (from: PublicKey, amount: UInt64) => AccountUpdate;
+In order to allow flexibility in granting permisssions, the methods in this section will call to a token admin contract, which is set during deployment of the token contract. That admin contract can grant or deny the right to mint tokens, pause/resume transfers, or change the admin contract itself. By using a third contract, the permissions can be changed without changing the token contract itself -- which is important for integration.
+
+#### The Admin Contract
+```TypeScript
+export type FungibleTokenAdminBase = SmartContract & {
+  canMint(accountUpdate: AccountUpdate): Promise<Bool>
+  canChangeAdmin(admin: PublicKey): Promise<Bool>
+  canPause(): Promise<Bool>
+  canResume(): Promise<Bool>
 }
 ```
 
-### Viewable
-For retrieving information about the token, such as the total or circulating supply, account balances, etc., we define the `Viewable` interface.
+An admin contract needs to provide the methods defined in `FungibleTokenAdminBase`, thus implementing that interface. Each of the methods will be called by the token contract to check permissions, and will return a `Bool` value to grant or deny permission.
+
+An example implementation that allows a priviledged key to perform any kind of administrative action is provided with the standard implementation.
 
+#### Deploying the Contract
 ```TypeScript
-interface Viewable {
-  getAccountOf: (address: PublicKey) => ReturnType<typeof Account>;
-  getBalanceOf: (address: PublicKey, options: ViewableOptions) => UInt64;
-  getTotalSupply: (options: ViewableOptions) => UInt64;
-  getCirculatingSupply: (options: ViewableOptions) => UInt64;
-  getDecimals: () => UInt64;
+export interface FungibleTokenDeployProps extends Exclude<DeployArgs, undefined> {
+  /** Address of the contract controlling permissions for administrative actions */
+  admin: PublicKey
+  /** The token symbol. */
+  symbol: string
+  /** A source code reference, which is placed within the `zkappUri` of the contract account. */
```

Comment:
Do we recommend developers where this url should point to? github?

Reply:
Github makes sense, I'll add a sentence.